tests/test_read_backed_experimental.py

```diff
@@ -0,0 +1,190 @@
+        "obs/int64"
+    )
+    assert store.get_access_count("var/int64") == 0, store.get_subkeys_accessed(
+        "var/int64"
+    )
+    # all codes read in for subset
+    assert store.get_access_count("obs/cat/codes") == 4, store.get_subkeys_accessed(
+        "obs/cat/codes"
+    )
+    remote[0:10, :].obs["int64"][0:10].compute()
+    assert store.get_access_count("obs/int64") == 1, store.get_subkeys_accessed(
+        "obs/int64"
+    )
+    # one for 0, .zmetadata handles .zarray
+    assert store.get_access_count("var/int64") == 0, store.get_subkeys_accessed(
+        "var/int64"
+    )  # never accessed
+
+
+@needs_xarray
+def test_access_count_dtype(adata_remote_with_store_tall_skinny):
+    remote, store = adata_remote_with_store_tall_skinny
+    store.initialize_key_trackers(["obs/cat/categories"])
+    assert (
+        store.get_access_count("obs/cat/categories") == 0
+    ), store.get_subkeys_accessed("obs/cat/categories")
+    # This should only cause categories to be read in once
+    remote.obs["cat"].dtype
+    remote.obs["cat"].dtype
+    remote.obs["cat"].dtype
+    assert (
+        store.get_access_count("obs/cat/categories") == 1
+    ), store.get_subkeys_accessed("obs/cat/categories")
+
+
+@needs_xarray
+def test_to_memory(adata_remote_orig):
+    remote, orig = adata_remote_orig
+    remote_to_memory = remote.to_memory()
+    assert_equal(remote_to_memory, orig)
+
+
+@needs_xarray
+def test_view_to_memory(adata_remote_orig):
+    remote, orig = adata_remote_orig
+    subset_obs = orig.obs["obs_cat"] == "a"
+    assert_equal(orig[subset_obs, :], remote[subset_obs, :].to_memory())
+
+    subset_var = orig.var["var_cat"] == "a"
+    assert_equal(orig[:, subset_var], remote[:, subset_var].to_memory())
+
+
+@needs_xarray
+def test_view_of_view_to_memory(adata_remote_orig):
+    remote, orig = adata_remote_orig
+    subset_obs = (orig.obs["obs_cat"] == "a") | (orig.obs["obs_cat"] == "b")
+    subsetted_adata = orig[subset_obs, :]
+    subset_subset_obs = subsetted_adata.obs["obs_cat"] == "b"
+    subsetted_subsetted_adata = subsetted_adata[subset_subset_obs, :]
+    assert_equal(
+        subsetted_subsetted_adata,
+        remote[subset_obs, :][subset_subset_obs, :].to_memory(),
+    )
+
+    subset_var = (orig.var["var_cat"] == "a") | (orig.var["var_cat"] == "b")
+    subsetted_adata = orig[:, subset_var]
+    subset_subset_var = subsetted_adata.var["var_cat"] == "b"
+    subsetted_subsetted_adata = subsetted_adata[:, subset_subset_var]
+    assert_equal(
+        subsetted_subsetted_adata,
+        remote[:, subset_var][:, subset_subset_var].to_memory(),
+    )
+
+
+@needs_xarray
+def test_unconsolidated(tmp_path, mtx_format):
+    adata = gen_adata((1000, 1000), mtx_format)
+    base_pth = Path(tmp_path)
+    orig_pth = base_pth / "orig.zarr"
+    write = lambda x: getattr(x, "write_zarr")(orig_pth)
+    write(adata)
+    (Path(orig_pth) / ".zmetadata").unlink()
```

Comment:
you don’t seem to trust that `Path`s stay `Path`s haha

```suggestion
    orig_pth = tmp_pth / "orig.zarr"
    adata.write_zarr(orig_pth)
    (orig_pth / ".zmetadata").unlink()
```